File: demo6-microservices/sendMessageService/src/main/java/com/infobip/sendMessageService/model/Message.java
package com.infobip.sendMessageService.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "messages")
@Getter
@Setter
//@Builder
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idMessage;
    @ManyToOne(cascade = {CascadeType.MERGE})
    @JoinColumn
    private User user;
//    private int userId;
    private String number;
    private Date dateTime;
    private String messageText;

    public Message() {

    }
}

File: sms-api-kotlin/sendmessageservice/src/main/kotlin/com/example/sendmessageservice/model/Message.kt
package com.example.sendmessageservice.model

import com.example.sendmessageservice.controller.dto.MessageDto
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "messages")
class Message (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idMessage: Long? = null,
    var number: String? = null,
    var dateTime: Date? = null,
    var messageText: String? = null,
    //    @JoinColumn(name = "user_idUser")
    @ManyToOne
    var userId: User? = null
)

fun Message.toMessageDto() = MessageDto(
    idMessage = idMessage,
    number = number,
    dateTime = dateTime,
    messageText = messageText,
    userId= userId?.idUser
)